x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/Authentication.java

```diff
@@ -282,6 +284,61 @@ private static Map<String, Object> maybeRewriteMetadata(TransportVersion olderVe
         }
     }
 
+    /**
+     * Creates a copy of this Authentication instance, but only with metadata entries specified by `fieldsToKeep`.
+     * All other entries are removed from the copy's metadata.
+     */
+    public Authentication copyWithFilteredMetadataFields(final Set<String> fieldsToKeep) {
+        Objects.requireNonNull(fieldsToKeep);
```

Comment:
I decided to keep this generic, and non-restrictive. For instance, the method does not enforce that `fieldsToKeep` contains the API key ID metadata field for an API-key authentication. If fields are filtered out that "break" the authentication instance, this is caught by the internal consistency checks further down. It feels cleaner to rely on those, rather than attempt to duplicate them here.